iceoryx_dust/test/moduletests/test_container_fixed_position_container.cpp

```diff
@@ -0,0 +1,1708 @@
+
+    sut.emplace(531U);
+    auto it = sut.emplace(369U);
+    sut.erase(it);
+
+    EXPECT_THAT(it.to_index(), Eq(Sut::Index::FIRST + 1U));
+}
+
+TEST_F(FixedPositionContainer_test, OriginsFromReturnsTrueWhenIteratorOriginsFromContainer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "3d462756-8cc8-42be-af02-83361ec527e0");
+
+    auto it = sut.emplace(121U);
+
+    EXPECT_TRUE(it.origins_from(sut));
+}
+
+TEST_F(FixedPositionContainer_test, OriginsFromReturnsFalseWhenIteratorDoesNotOriginFromContainer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "cb400b16-b705-4512-8701-5cf62cf82534");
+
+    auto it = sut.emplace(213U);
+    Sut sut2;
+
+    EXPECT_FALSE(it.origins_from(sut2));
+}
+
+TEST_F(FixedPositionContainer_test, CompareForEqualityReturnsTrueWhenIteratorPointToTheSameElement)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "846b2153-ebcd-4810-b86a-56ba36b141e6");
+
+    auto it1 = sut.emplace(987U);
+    auto it2 = sut.begin();
+
+    EXPECT_TRUE(it1 == it2);
+}
+
+TEST_F(FixedPositionContainer_test, CompareForEqualityReturnsFalseWhenIteratorDoesNotPointToTheSameElement)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2801a787-bda2-4a6d-8cea-88ca7a59c075");
+
+    auto it1 = sut.emplace(963U);
+    auto it2 = sut.emplace(963U);
+
+    EXPECT_FALSE(it1 == it2);
+}
+
+TEST_F(FixedPositionContainer_test, CompareForEqualityReturnsFalseWhenIteratorDoesNotOriginFromTheSameContainer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8f4547de-5acc-458f-9cfc-95e472fa3732");
+
+    auto it1 = sut.emplace(842U);
+    Sut sut2;
+    auto it2 = sut2.emplace(842U);
+
+    EXPECT_FALSE(it1 == it2);
+}
+
+TEST_F(FixedPositionContainer_test, CompareForNonEqualityReturnsFalseWhenIteratorPointToTheSameElement)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "4301a29c-b90c-4dba-9431-1d668e2de2f5");
+
+    auto it1 = sut.emplace(987U);
+    auto it2 = sut.begin();
+
+    EXPECT_FALSE(it1 != it2);
+}
+
+TEST_F(FixedPositionContainer_test, CompareForNonEqualityReturnsTrueWhenIteratorDoesNotPointToTheSameElement)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a76db7ad-0d4e-4d15-92c8-9a5b53d431e9");
+
+    auto it1 = sut.emplace(963U);
+    auto it2 = sut.emplace(963U);
+
+    EXPECT_TRUE(it1 != it2);
+}
+
+TEST_F(FixedPositionContainer_test, CompareForNonEqualityReturnsTrueWhenIteratorDoesNotOriginFromTheSameContainer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "03d3bc8e-03da-4b1c-ac25-8606e132f7ac");
+
+    auto it1 = sut.emplace(842U);
+    Sut sut2;
+    auto it2 = sut2.emplace(842U);
+
+    EXPECT_TRUE(it1 != it2);
+}
+
+TEST_F(FixedPositionContainer_test, IteratorDestructorDoesNotDestroyObjectItPointsTo)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a52e9d6e-c763-4953-95c2-1f05d605d180");
+
+    fillSutComplex();
+
+    {
+        auto it IOX_MAYBE_UNUSED = sut_complex.begin();
+        auto cit IOX_MAYBE_UNUSED = sut_complex.cbegin();
+    }
+
+    EXPECT_THAT(stats.dTor, Eq(0));
+}
```

Comment:
Wow.. These tests compete with the string tests :stuck_out_tongue_winking_eye:

Reply:
Nothing competes with the string tests :laughing: